pkg/archive/archive.go

```diff
@@ -382,6 +382,56 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 	return nil
 }
 
+// Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
+func Matches(relFilePath string, patterns []string) (bool, error) {
+
+	if filepath.Clean(relFilePath) == "." {
+		// Don't let them exclude everything, kind of silly.
+		return false, nil
+	}
+
+	matched := false
+	parentPath := filepath.Dir(relFilePath)
+	parentPathDirs := strings.Split(parentPath, "/")
+
+	for _, pattern := range patterns {
+		negative := false
+		if fileutils.Empty(pattern) {
```

Comment:
Let's remove empty patterns from your `cleanedPattern` slice - then we can remove this if-statement because we know the list won't contain them.